packages/web3-core/src/web3_config.ts

```diff
@@ -255,20 +322,64 @@ export abstract class Web3Config
 		this._config.defaultChain = val;
 	}
 
+	/**
+	 * The default hardfork property is used for signing transactions locally.
+	 *
+	 * Will return the default hardfork
+	 *
+	 * @returns {string}
+	 *
+	 * The default hardfork property can be one of the following:
+	 * - `chainstart`
+	 * - `homestead`
+	 * - `dao`
+	 * - `tangerineWhistle`
+	 * - `spuriousDragon`
+	 * - `byzantium`
+	 * - `constantinople`
+	 * - `petersburg`
+	 * - `istanbul`
+	 * - `berlin`
+	 * - `london`
+	 *
+	 * Default is `london`
+	 */
 	public get defaultHardfork() {
 		return this._config.defaultHardfork;
 	}
 
+	/**
+	 *
+	 * @param {string} val
```

Comment:
We don't need to document the types, it will be picked by Typescript. 

```suggestion
	 * @param val - Set the value for the hardfork
```